Stork environment variable resolution not working properly

### Describe the bug

When trying to use Stork and trying to set properties (i.e. the `service-discovery.address-list` property) it doesn't seem to resolve properly.

### Expected behavior

I would expect to be able to set Stork properties via env variables.

### Actual behavior

It doesn't seem that I can express `quarkus.stork.vehicle-data-service.service-discovery.address-list=vpic.nhtsa.dot.gov` as an environment variable.

### How to Reproduce?

Reproducer: 
[stork-broken.zip](https://github.com/user-attachments/files/15918554/stork-broken.zip)

1. Download and unzip
2. `cd stork-broken`
3. `./mvnw clean package`
4. `export QUARKUS_STORK_VEHICLE_DATA_SERVICE_SERVICE_DISCOVERY_ADDRESS_LIST=vpic.nhtsa.dot.gov`
5. `java -jar target/quarkus-app/quarkus-run.jar`

You get the following error:

```
java.util.NoSuchElementException: SRCFG00014: The config property quarkus.stork.vehicle.service-discovery.type is required but it could not be found in any config source
        at io.smallrye.config.SmallRyeConfig.convertValue(SmallRyeConfig.java:435)
        at io.smallrye.config.SmallRyeConfig.getValue(SmallRyeConfig.java:380)
        at io.smallrye.config.ConfigMappingContext$ObjectCreator$9.apply(ConfigMappingContext.java:670)
        at io.smallrye.config.ConfigMappingContext$ObjectCreator$9.apply(ConfigMappingContext.java:665)
        at io.smallrye.config.ConfigMappingContext$ObjectCreator$1.accept(ConfigMappingContext.java:414)
        at io.smallrye.config.ConfigMappingContext$ObjectCreator$1.accept(ConfigMappingContext.java:411)
        at io.smallrye.config.ConfigMappingContext$ObjectCreator.value(ConfigMappingContext.java:665)
        at io.quarkus.stork.StorkServiceDiscoveryConfiguration1376039613Impl.<init>(Unknown Source)
        at java.base/jdk.internal.reflect.DirectConstructorHandleAccessor.newInstance(DirectConstructorHandleAccessor.java:62)
        at java.base/java.lang.reflect.Constructor.newInstanceWithCaller(Constructor.java:502)
        at java.base/java.lang.reflect.Constructor.newInstance(Constructor.java:486)
        at io.smallrye.config.ConfigMappingLoader.configMappingObject(ConfigMappingLoader.java:108)
        at io.smallrye.config.ConfigMappingContext.constructGroup(ConfigMappingContext.java:93)
        at io.smallrye.config.ConfigMappingContext$ObjectCreator$6.apply(ConfigMappingContext.java:610)
        at io.smallrye.config.ConfigMappingContext$ObjectCreator$6.apply(ConfigMappingContext.java:604)
        at io.smallrye.config.ConfigMappingContext$ObjectCreator$1.accept(ConfigMappingContext.java:414)
        at io.smallrye.config.ConfigMappingContext$ObjectCreator$1.accept(ConfigMappingContext.java:411)
        at io.smallrye.config.ConfigMappingContext$ObjectCreator.group(ConfigMappingContext.java:604)
        at io.quarkus.stork.ServiceConfiguration2122640426Impl.<init>(Unknown Source)
        at java.base/jdk.internal.reflect.DirectConstructorHandleAccessor.newInstance(DirectConstructorHandleAccessor.java:62)
        at java.base/java.lang.reflect.Constructor.newInstanceWithCaller(Constructor.java:502)
        at java.base/java.lang.reflect.Constructor.newInstance(Constructor.java:486)
        at io.smallrye.config.ConfigMappingLoader.configMappingObject(ConfigMappingLoader.java:108)
        at io.smallrye.config.ConfigMappingContext.constructGroup(ConfigMappingContext.java:93)
        at io.smallrye.config.ConfigMappingContext$ObjectCreator$7.apply(ConfigMappingContext.java:628)
        at io.smallrye.config.ConfigMappingContext$ObjectCreator$7.apply(ConfigMappingContext.java:621)
        at io.smallrye.config.ConfigMappingContext$ObjectCreator$3$4.accept(ConfigMappingContext.java:532)
        at io.smallrye.config.ConfigMappingContext$ObjectCreator$3$4.accept(ConfigMappingContext.java:514)
        at io.smallrye.config.ConfigMappingContext$ObjectCreator.lazyGroup(ConfigMappingContext.java:621)
        at io.quarkus.stork.StorkConfiguration-1155448456Impl.<init>(Unknown Source)
        at java.base/jdk.internal.reflect.DirectConstructorHandleAccessor.newInstance(DirectConstructorHandleAccessor.java:62)
        at java.base/java.lang.reflect.Constructor.newInstanceWithCaller(Constructor.java:502)
        at java.base/java.lang.reflect.Constructor.newInstance(Constructor.java:486)
        at io.smallrye.config.ConfigMappingLoader.configMappingObject(ConfigMappingLoader.java:108)
        at io.smallrye.config.ConfigMappingContext.constructGroup(ConfigMappingContext.java:93)
        at io.smallrye.config.ConfigMappingContext.constructRoot(ConfigMappingContext.java:88)
        at io.smallrye.config.ConfigMappingContext.<init>(ConfigMappingContext.java:81)
        at io.smallrye.config.ConfigMappingContext.<init>(ConfigMappingContext.java:50)
        at io.smallrye.config.SmallRyeConfig$1.get(SmallRyeConfig.java:128)
        at io.smallrye.config.SmallRyeConfig$1.get(SmallRyeConfig.java:125)
        at io.smallrye.config.SecretKeys.doUnlocked(SecretKeys.java:28)
        at io.smallrye.config.SmallRyeConfig.buildMappings(SmallRyeConfig.java:125)
        at io.smallrye.config.SmallRyeConfig.<init>(SmallRyeConfig.java:88)
        at io.smallrye.config.SmallRyeConfigBuilder.build(SmallRyeConfigBuilder.java:691)
        at io.quarkus.runtime.generated.Config.readConfig(Unknown Source)
        at io.quarkus.runtime.generated.Config.createRunTimeConfig(Unknown Source)
        at io.quarkus.deployment.steps.RuntimeConfigSetup.deploy(Unknown Source)
        at io.quarkus.runner.ApplicationImpl.doStart(Unknown Source)
        at io.quarkus.runtime.Application.start(Application.java:101)
        at io.quarkus.runtime.ApplicationLifecycleManager.run(ApplicationLifecycleManager.java:111)
        at io.quarkus.runtime.Quarkus.run(Quarkus.java:71)
        at io.quarkus.runtime.Quarkus.run(Quarkus.java:44)
        at io.quarkus.runtime.Quarkus.run(Quarkus.java:124)
        at io.quarkus.runner.GeneratedMain.main(Unknown Source)
        at java.base/jdk.internal.reflect.DirectMethodHandleAccessor.invoke(DirectMethodHandleAccessor.java:103)
        at java.base/java.lang.reflect.Method.invoke(Method.java:580)
        at io.quarkus.bootstrap.runner.QuarkusEntryPoint.doRun(QuarkusEntryPoint.java:62)
        at io.quarkus.bootstrap.runner.QuarkusEntryPoint.main(QuarkusEntryPoint.java:33)
Configuration validation failed:
        java.util.NoSuchElementException: SRCFG00014: The config property quarkus.stork.vehicle.service-discovery.type is required but it could not be found in any config source
java.lang.RuntimeException: Failed to start quarkus
        at io.quarkus.runner.ApplicationImpl.doStart(Unknown Source)
        at io.quarkus.runtime.Application.start(Application.java:101)
        at io.quarkus.runtime.ApplicationLifecycleManager.run(ApplicationLifecycleManager.java:111)
        at io.quarkus.runtime.Quarkus.run(Quarkus.java:71)
        at io.quarkus.runtime.Quarkus.run(Quarkus.java:44)
        at io.quarkus.runtime.Quarkus.run(Quarkus.java:124)
        at io.quarkus.runner.GeneratedMain.main(Unknown Source)
        at java.base/jdk.internal.reflect.DirectMethodHandleAccessor.invoke(DirectMethodHandleAccessor.java:103)
        at java.base/java.lang.reflect.Method.invoke(Method.java:580)
        at io.quarkus.bootstrap.runner.QuarkusEntryPoint.doRun(QuarkusEntryPoint.java:62)
        at io.quarkus.bootstrap.runner.QuarkusEntryPoint.main(QuarkusEntryPoint.java:33)
Caused by: io.quarkus.runtime.configuration.ConfigurationException: Failed to read configuration properties
        at io.quarkus.deployment.steps.RuntimeConfigSetup.deploy(Unknown Source)
        ... 11 more
Caused by: io.smallrye.config.ConfigValidationException: Configuration validation failed:
        java.util.NoSuchElementException: SRCFG00014: The config property quarkus.stork.vehicle.service-discovery.type is required but it could not be found in any config source
        at io.smallrye.config.SmallRyeConfig.buildMappings(SmallRyeConfig.java:138)
        at io.smallrye.config.SmallRyeConfig.<init>(SmallRyeConfig.java:88)
        at io.smallrye.config.SmallRyeConfigBuilder.build(SmallRyeConfigBuilder.java:691)
        at io.quarkus.runtime.generated.Config.readConfig(Unknown Source)
        at io.quarkus.runtime.generated.Config.createRunTimeConfig(Unknown Source)
        ... 12 more
```

The value it _should_ be trying to set is `quarkus.stork.vehicle-data-service.service-discovery.address-list=vpic.nhtsa.dot.gov`.

You can also try

1. `export QUARKUS_STORK__VEHICLE_DATA_SERVICE__SERVICE_DISCOVERY_ADDRESS_LIST=vpic.nhtsa.dot.gov`
2. `java -jar target/quarkus-app/quarkus-run.jar`

But that doesn't work either. It has the same error.

### Output of `uname -a` or `ver`

`Darwin edeandrea-m1pro 23.5.0 Darwin Kernel Version 23.5.0: Wed May  1 20:12:58 PDT 2024; root:xnu-10063.121.3~5/RELEASE_ARM64_T6000 arm64`

### Output of `java -version`

```
openjdk version "17.0.11" 2024-04-16
OpenJDK Runtime Environment Temurin-17.0.11+9 (build 17.0.11+9)
OpenJDK 64-Bit Server VM Temurin-17.0.11+9 (build 17.0.11+9, mixed mode)
```

### Quarkus version or git rev

3.11.3

### Build tool (ie. output of `mvnw --version` or `gradlew --version`)

```
Apache Maven 3.9.6 (bc0240f3c744dd6b6ec2920b3cd08dcc295161ae)
Maven home: /Users/edeandre/.m2/wrapper/dists/apache-maven-3.9.6-bin/3311e1d4/apache-maven-3.9.6
Java version: 17.0.11, vendor: Eclipse Adoptium, runtime: /Users/edeandre/.sdkman/candidates/java/17.0.11-tem
Default locale: en_US, platform encoding: UTF-8
OS name: "mac os x", version: "14.5", arch: "aarch64", family: "mac"
```

### Additional information

If you run the app in dev mode it should work because I have `%dev.quarkus.stork.vehicle-data-service.service-discovery.address-list=vpic.nhtsa.dot.gov` in `application.properties`.

To see how it should behave, when running dev mode you can `curl http://localhost:8080/vehicle-data/audi`. You should get something back like this:

```json
{
    "Count": 2,
    "Message": "Response returned successfully",
    "Results": [
        {
            "MakeId": 582,
            "MakeName": "AUDI",
            "VehicleTypeId": 2,
            "VehicleTypeName": "Passenger Car"
        },
        {
            "MakeId": 582,
            "MakeName": "AUDI",
            "VehicleTypeId": 7,
            "VehicleTypeName": "Multipurpose Passenger Vehicle (MPV)"
        }
    ],
    "SearchCriteria": "Make: audi"
}
```

--- a/bom/application/pom.xml
+++ b/bom/application/pom.xml
@@ -51,7 +51,7 @@
         <microprofile-lra.version>2.0</microprofile-lra.version>
         <microprofile-openapi.version>3.1.1</microprofile-openapi.version>
         <smallrye-common.version>2.3.0</smallrye-common.version>
-        <smallrye-config.version>3.8.2</smallrye-config.version>
+        <smallrye-config.version>3.8.3</smallrye-config.version>
         <smallrye-health.version>4.1.0</smallrye-health.version>
         <smallrye-metrics.version>4.0.0</smallrye-metrics.version>
         <smallrye-open-api.version>3.10.0</smallrye-open-api.version>

